test/uk/gov/hmrc/excisemovementcontrolsystemapi/service/MovementServiceSpec.scala

```diff
@@ -378,131 +378,231 @@ class MovementServiceSpec extends PlaySpec with EitherValues with BeforeAndAfter
 
-      setUpForUpdateMovement(newMessage, Seq(None), Some("123"), "<foo>test</foo>", cachedMovements)
+      setUpForUpdateMovement(newMessage, Seq(None), Some("123"), "<foo>test</foo>", cachedMovements, messageIdForNewMessage)
       when(mockMovementRepository.getAllBy(any)).thenReturn(Future.successful(Seq(movementWithMessagesAlready)))
 
-      await(movementServiceForUpdateTests.updateMovement(newMessage, consignorId))
+      await(movementService.updateMovement(newMessage, consignorId))
 
       verify(mockMovementRepository).updateMovement(eqTo(movementWithMessagesAlready))
     }
 
     "save to DB when message has different content but the same message type" in {
-      val cachedMessage: Message = createMessage("<foo>different content</foo>", MessageTypes.IE801.value)
+      val cachedMessage: Message = createMessage("<foo>different content</foo>", MessageTypes.IE801.value, messageIdForNewMessage)
       val movementWithMessagesAlready = Movement("boxId", lrn, consignorId, None, None, now, Seq(cachedMessage, cachedMessage1))
 
-      setUpForUpdateMovement(newMessage, Seq(None), Some("123"), "<foo>test</foo>", cachedMovements)
-      when(mockMovementRepository.updateMovement(any)).thenReturn(Future.successful(true))
+      setUpForUpdateMovement(newMessage, Seq(None), Some("123"), "<foo>test</foo>", cachedMovements, messageIdForNewMessage)
       when(mockMovementRepository.getAllBy(any)).thenReturn(Future.successful(Seq(movementWithMessagesAlready)))
 
-      await(movementServiceForUpdateTests.updateMovement(newMessage, consignorId))
+      await(movementService.updateMovement(newMessage, consignorId))
 
-      val expectedNewMessage = createMessage("<foo>test</foo>", MessageTypes.IE818.value)
+      val expectedNewMessage = createMessage("<foo>test</foo>", MessageTypes.IE818.value, messageIdForNewMessage)
       val expectedMovement = movementWithMessagesAlready.copy(messages = Seq(cachedMessage, cachedMessage1, expectedNewMessage))
       verify(mockMovementRepository).updateMovement(eqTo(expectedMovement))
     }
 
-    "return false if did not save the message" in {
-      setUpForUpdateMovement(newMessage, Seq(None), Some("123"), "<foo>test</foo>", cachedMovements)
-      when(mockMovementRepository.updateMovement(any)).thenReturn(Future.successful(false))
+    "return an empty sequence if did not save the message" in {
+      setUpForUpdateMovement(newMessage, Seq(None), Some("123"), "<foo>test</foo>", cachedMovements, messageIdForNewMessage)
+      when(mockMovementRepository.updateMovement(any))
+        .thenReturn(Future.successful(None))
 
-      val result = await(movementServiceForUpdateTests.updateMovement(newMessage, consignorId))
+      val result = await(movementService.updateMovement(newMessage, consignorId))
 
-      result mustBe false
+      result mustBe Seq.empty
     }
+
+//    "getMovementByArcOrByErnAndLrn" should {
```

Comment:
Why are all these tests commented?

Reply:
That production code does not exist. This was initial implementation, but I removed later because I change mind on the implementation and forgot to remove it. Removed now